en_US/data-integration/schema-registry-example-protobuf.md

```diff
@@ -0,0 +1,185 @@
+   - If you have changed EMQX's default configuration. For example, the configuration change on authentication can require you to type in a username and password.
+
+3. Click **Connect** to connect to the EMQX instance as an MQTT client.
+4. In the **Subscription** area, type `person/#` in the **Topic** field and click **Subscribe**.
+
+5. Install the Python dependencies and execute the device-side code:
+
+   ```shell
+   $ pip3 install protobuf paho-mqtt
+   $ protoc --python_out=. person.proto
+   
+   $ python3 protobuf_mqtt.py
+   Connected with result code 0
+   publish to topic: t/1, payload: b'\n\x05Shawn\x10\x01\x1a\x11shawn@example.com'
+   ```
+
+6. Check that a message with the topic `person/Shawn` is received on the Websocket side:
+
+   ```json
+   {"name":"Shawn","id":1,"email":"shawn@example.com"}
+   ```
+
+## Encoding Scenario
+
+A device subscribes to a topic `protobuf_out` expecting a binary message encoded using Protobuf. The rule engine is used to encode such a message and publish it to the associated topic.
+
+### Create Schema
+
+Use the same schema as described in the [decoding scenario](#decoding-scenario).
+
+### Create the rule
+
+1. In the Dashboard, select **Integration** -> **Rules** from the navigation menu.
+
+2. On the **Rules** page, click **Create** at the top right corner.
+
+3. Use the schema you have just created to write the rule SQL statement:
+
+   ```sql
+   SELECT
+     schema_encode('protobuf_person', json_decode(payload), 'Person') as protobuf_person
+   FROM
+     "protobuf_in"
+   ```
+
+   The key point here is `schema_encode('protobuf_person', json_decode(payload), 'Person')`:
+
+   - The `schema_encode` function encodes the contents of the payload field according to the schema `protobuf_person`;
+   - `as person` stores the encoded value in the variable `person`;
+   - The last argument `Person` specifies that the message type in the payload is the `Person` type defined in the Protobuf schema.
+   - `json_decode(payload)` is needed because `payload` is generally a JSON-encoded binary, and `schema_encode` requires a Map as its input.
+
+4. Click **Add Action**.  Select `Republish` from the drop-down list of the **Action** field. 
+5. In the **Topic** field, type `protobuf_out` as the destination topic.
+6. In the **Payload** field, type message content template: `${protobuf_person}`.
+
+This action sends the Protobuf encoded user message to the topic `protobuf_out`. `${protobuf_person}` is a variable placeholder that will be replaced at runtime with the value of the result of `schema_encode` (a binary value).
+
+### Prepare Device-Side Code
+
+Once the rules have been created, you can simulate the data for testing.
+
+The following code uses the Python language to fill a user message, encode it as binary data, then send it to the `protobuf_in` topic. See [full code](https://gist.github.com/thalesmg/c5f03f99f982401d16ef6583e30144fa) for details.
```

Comment:
I just have a doubt here. In the encoding scenario, the message still needs to be encoded as binary data in this step? Is that correct? @thalesmg

Reply:
Yes, it is encoded as binary here.  I've added a print statement to print it out.